File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.py
def solution(n, arr1, arr2):
    answer = []
    # arr의 각 원소들 꺼내서 2진수 or 연산 -> 새로운 배열 만들기
    for i in range(n):
        answer.append(bin(arr1[i]|arr2[i])[2:])
    # 그 배열의 1을 replace #
    for i in range(n):
        while len(answer[i]) < n:
            answer[i] = ' ' + answer[i]
        answer[i] = answer[i].replace('1', '#')
        answer[i] = answer[i].replace('0', ' ')
        
    return answer

File: 프로그래머스/1/17681. ［1차］ 비밀지도/［1차］ 비밀지도.swift
import Foundation

func solution(_ n:Int, _ arr1:[Int], _ arr2:[Int]) -> [String] {
    let entireMap = (0..<n)
        .map { arr1[$0] | arr2[$0] }
        .map {
            String($0, radix: 2)
                .replacingOccurrences(of: "0", with: " ")
                .replacingOccurrences(of: "1", with: "#")
        }
    
    
    let paddedMap = entireMap.map { String(repeating: " ", count: n - $0.count) + $0 }
    
    return paddedMap
}